azure-quantum/azure/quantum/target/quantinuum.py

```diff
@@ -160,9 +160,8 @@ def estimate_cost(
 
             else:
                 from qiskit.dagcircuit.dagnode import DAGOpNode
-                qasm = Qasm(data=circuit)
-                ast = qasm.parse()
-                dag = ast_to_dag(ast)
+                circuit_obj = loads(string=circuit)
+                dag = circuit_to_dag(circuit=circuit_obj)
```

Comment:
For now, it seems like it works even without parsing the node for the circuit program (`ast = qasm.parse()`), and just calling `circuit_to_dag` is enough for getting the same `DAGCircuit` object, but I am still not sure, so need to double-check.